src/hooks/useExpenditureStaking.ts

```diff
@@ -0,0 +1,82 @@
+import { Extension, Id } from '@colony/colony-js';
+import { BigNumber } from 'ethers';
+import { useEffect, useState } from 'react';
+
+import { ADDRESS_ZERO } from '~constants';
+import { useAppContext } from '~context/AppContext.tsx';
+import { useColonyContext } from '~context/ColonyContext.tsx';
+import { useGetUserReputationQuery } from '~gql';
+import useEnoughTokensForStaking from '~hooks/useEnoughTokensForStaking.ts';
+import useExtensionData from '~hooks/useExtensionData.ts';
+import { isInstalledExtensionData } from '~utils/extensions.ts';
+
+const useExpenditureStaking = (createdInDomainId = Id.RootDomain) => {
+  const { colony } = useColonyContext();
+  const { user } = useAppContext();
+  const { walletAddress = '' } = user ?? {};
+
+  const [stakeAmount, setStakeAmount] = useState<string>();
+
+  const { data, loading: reputationLoading } = useGetUserReputationQuery({
+    fetchPolicy: 'network-only',
+    variables: {
+      input: {
+        colonyAddress: colony.colonyAddress,
+        walletAddress: ADDRESS_ZERO,
+        domainId: createdInDomainId,
+      },
+    },
+  });
+  const totalDomainReputation = data?.getUserReputation;
+
+  const { extensionData, loading: extensionLoading } = useExtensionData(
+    Extension.StakedExpenditure,
+  );
+
+  const { loadingUserTokenBalance, hasEnoughTokens } =
+    useEnoughTokensForStaking(
+      colony.nativeToken.tokenAddress,
+      walletAddress,
+      colony.colonyAddress,
+      stakeAmount ?? '0',
+    );
+
+  useEffect(() => {
+    if (
+      !totalDomainReputation ||
+      !extensionData ||
+      !isInstalledExtensionData(extensionData)
+    ) {
+      return;
+    }
+
+    const { stakeFraction } = extensionData.params?.stakedExpenditure || {};
+    if (!stakeFraction) {
+      return;
+    }
+
+    const requiredAmount = BigNumber.from(stakeFraction)
+      .mul(totalDomainReputation)
+      .div(BigNumber.from(10).pow(18))
```

Comment:
is the pow(18) gonna be always a thing or do we want the staking token decimals here?

Reply:
great catch, it should be native token decimals